### PR TITLE
etcd: Add API to delete all about the key.

### PR DESCRIPTION
**Pull request by xiekeyi98**
What problem does this PR solve?

Add a new API so that we can delete all about the key inputted by only one command.

What is changed and how it works?

Add a flag `-all` to delete all keys about the key.

Check List

- Unit test

### CODE REVIEW

**Comment by xiekeyi98** [comment]
test

**Comment by xiekeyi98** [comment]
Because discuss in the Slack, I will change it.

**Comment by xiekeyi98** [comment]
image

test

File: cmd/etcd.go [hunk 1]
```diff
@@ -150,24 +144,68 @@ func putKeyCommandFunc(cmd *cobra.Command, args []string) {
...
-		cmd.Printf("Failed to put key: %v\n", err)
+		cmd.Println(errors.Wrap(err, "Failed to put key"))
 		return
 	}
 
 	cmd.Println(res)
 }
 
+func evictOwnerCommandFunc(cmd *cobra.Command, args []string) {
+	cmd.Println("[DANGEROUS] Are you sure to evict the owner?yes/no.")
```
Comment (winkyao) [inline cmd/etcd.go]: `[DANGEROUS] Are you sure to evict the owner? Please type [yes/no].`

File: cmd/etcd.go [hunk 2]
```diff
@@ -150,24 +144,68 @@ func putKeyCommandFunc(cmd *cobra.Command, args []string) {
...
+		if strings.HasPrefix(v["key"], ddlOwnerKeyPrefix) {
+			ddlOwnerKey = v["key"]
+			ddlOwnerSchemaVersionKey = ddlAllSchemaVersionsPrefix + v["value"]
+			break
+		}
+	}
+
+	_, err = delKey(ddlOwnerKey)
+	if err != nil {
+		cmd.Println(errors.Wrap(err, "Failed to evict"))
```
Comment (winkyao) [inline cmd/etcd.go]: fmt.Sprintf("Failed to delete the key:%v", ddlOwnerKey)

File: cmd/etcd.go [hunk 3]
```diff
@@ -150,24 +144,68 @@ func putKeyCommandFunc(cmd *cobra.Command, args []string) {
...
+		return
+	}
+	_, err = delKey(ddlOwnerSchemaVersionKey)
+	if err != nil {
+		cmd.Println(errors.Wrap(err, "Failed to evict"))
```
Comment (winkyao) [inline cmd/etcd.go]: fmt.Sprintf("Failed to delete the key:%v", ddlOwnerSchemaVersionKey)

File: cmd/etcd.go (hunk 2 above)
Comment (xiekeyi98) [reply cmd/etcd.go]: `Sprintf`?
I have changed it to `cmd.Printf`.

**Review by winkyao — COMMENTED** [review]
LGTM

**Comment by xiekeyi98** [comment]
@zimulala  @crazycs520  PTAL.

**Review by crazycs520 — COMMENTED** [review]
LGTM

**Review by crazycs520 — APPROVED** [review]

File: cmd/etcd.go [hunk 4]
```diff
@@ -75,56 +82,43 @@ func newDelKeyCommand() *cobra.Command {
...
+		Short: "[ONLY FOR TEST] put a key in the path of TiDB schema versions by `putkey [key] [value]`",
 		Run:   putKeyCommandFunc,
 	}
 	return m
 }
 
+// newEvictOwnerCommand returns a evict owner subcommand of EtcdCommand.
+func newEvictOwnerCommand() *cobra.Command {
+	m := &cobra.Command{
+		Use:   "evictowner",
```
Comment (zimulala) [inline cmd/etcd.go:95]: Could we update this comment?
Maybe this key isn't the owner, it's only a DDL object information.

**Review by zimulala — CHANGES_REQUESTED** [review]

File: cmd/etcd.go (hunk 4 above)
Comment (xiekeyi98) [reply cmd/etcd.go:95]: Why this key is not the owner?
IMO, this key must be the owner.

Comment (zimulala) [reply cmd/etcd.go:95]: This key's path prefix has "owner"，it uses to campaign the owner. But it doesn't mean that the key with this prefix is the owner.
3 TiDB nodes:
```
# ETCDCTL_API=3 etcdctl --endpoints=127.0.0.1:42379 get /tidb/ddl  --prefix
/tidb/ddl/all_schema_versions/1ade5078-57d9-439f-bb29-0fc9daa3c095
15
/tidb/ddl/all_schema_versions/27509709-fe10-492b-9c37-19aea6492233
15
/tidb/ddl/all_schema_versions/aa6b2dc0-922c-4b5d-885b-eb64907002a5
15
/tidb/ddl/fg/owner/12c8697feef62d5f
27509709-fe10-492b-9c37-19aea6492233
/tidb/ddl/fg/owner/12c8697feef62d73
aa6b2dc0-922c-4b5d-885b-eb64907002a5
/tidb/ddl/fg/owner/12c8697feef62d85
1ade5078-57d9-439f-bb29-0fc9daa3c095
/tidb/ddl/global_schema_version
0
```

File: cmd/etcd.go [hunk 5]
```diff
@@ -150,24 +144,68 @@ func putKeyCommandFunc(cmd *cobra.Command, args []string) {
...
 	if err != nil {
-		cmd.Printf("Failed to put key: %v\n", err)
+		cmd.Println(errors.Wrap(err, "Failed to put key"))
 		return
 	}
 
 	cmd.Println(res)
 }
 
+func evictOwnerCommandFunc(cmd *cobra.Command, args []string) {
```
Comment (zimulala) [inline cmd/etcd.go:165]: We don't use the `args` in this function, so how do we know which DDL we remove?

**Comment by xiekeyi98** [comment]
Because of https://github.com/pingcap/tidb-ctl/pull/18#discussion_r265859106 , this PR will be closed.
`./tidb-ctl delkey {key}` is enough now.

**Pull request closed without merging**